src/js/component/panel/PanelIssues.js

```diff
@@ -50,72 +67,195 @@ const defaultProps = {
-                || (props.filters.feature && isFeature);
+            return (filters.improvement && isImprovement)
+                || (filters.task && isTask)
+                || (filters.feature && isFeature);
         });
     }
 
-    if (!_.size(props.data) && props.hideOnEmpty) {
-        return null;
+    // Sort by priority, then owner
+    preparedIssues = _.sortBy(preparedIssues, (item) => {
+        const priority = priorities[item.url ?? ''] && (priorities[item.url].priority !== undefined)
+            ? priorities[item.url].priority
+            : Number.MAX_SAFE_INTEGER;
+        return [priority, item.currentUserIsOwner ? 0 : 1];
+    });
+
+    // Use localOrder if available, while waiting for Onyx to update
+    if (localOrder.length && localOrder.length === preparedIssues.length) {
+        const dataById = _.indexBy(preparedIssues, 'id');
+        return _.filter(_.map(localOrder, id => dataById[id]), Boolean);
     }
+    return preparedIssues;
+}
+
+function PanelIssues(props) {
+    const [priorities = {}] = useOnyx(`${ONYXKEYS.ISSUES.COLLECTION_PRIORITIES}${props.title}`);
+    const [activeId, setActiveId] = useState(null);
+
+    // Add local state for ordered issues so that it can be updated synchronously when the user drags an issue and drops it,
+    // preventing the item from jumping back to its original position briefly
+    const [localOrder, setLocalOrder] = useState([]);
 
-    // Put the issues owned by the current user at the top of the list
-    const sortedData = _.sortBy(filteredData, 'currentUserIsOwner');
+    // When priorities or filteredData change (i.e., Onyx updates), clear localOrder only if it's set
+    useEffect(() => {
+        if (!localOrder.length) {
+            return;
+        }
+        setLocalOrder([]);
+    }, [priorities, props.data, props.filters, props.hideIfHeld, props.hideIfUnderReview, props.hideIfOwnedBySomeoneElse, props.applyFilters, localOrder]);
+
+    const filteredData = useMemo(() => getOrderedFilteredIssues({
+        issues: props.data,
+        filters: props.filters,
+        priorities,
+        localOrder,
+        hideIfHeld: props.hideIfHeld,
+        hideIfUnderReview: props.hideIfUnderReview,
+        hideIfOwnedBySomeoneElse: props.hideIfOwnedBySomeoneElse,
+        applyFilters: props.applyFilters,
+    }), [
+        props.data,
+        props.hideIfHeld,
+        props.hideIfUnderReview,
+        props.hideIfOwnedBySomeoneElse,
+        props.applyFilters,
+        props.filters,
+        priorities,
+        localOrder,
+    ]);
+
+    const sensors = useSensors(
+        useSensor(PointerSensor, {
+            activationConstraint: {
+                distance: 5,
+            },
+        }),
+    );
+
+    const issueIds = _.map(filteredData, issue => issue.id.toString());
+    const activeIssue = activeId ? _.find(filteredData, issue => issue.id.toString() === activeId) : null;
+
+    const handleDragStart = useCallback((event) => {
```

Comment:
Since methods should be named for what they do, not what they handle, this method should really be called `setActiveId`... but since `setActiveId()` is already a method, I think you can just call `setActiveId()` directly in the component attribute like `onDragStart={e => setActiveId(e.active.id)}`. I think these one-liners are a valid exception to having event handlers defined separately from the component.